### PR TITLE
SER-3150 | Allow to use unified-search in Special:Search

### PR DESCRIPTION
**Pull request by slayful**

### CODE REVIEW

File: extensions/wikia/Search/WikiaSearchController.class.php [hunk 1]
```diff
@@ -107,18 +110,148 @@ public function index() {
 			$this->setVarnishCacheTime( WikiaResponse::CACHE_STANDARD );
 		}
 
-		$this->handleLayoutAbTest( $this->getVal( 'ab', null ), $searchConfig->getNamespaces() );
-		if ( $searchConfig->getQuery()->hasTerms() ) {
-			$search = $this->queryServiceFactory->getFromConfig( $searchConfig );
-			/* @var $search Wikia\Search\QueryService\Select\Dismax\OnWiki */
-			$search->getMatch();
+		$this->handleLayoutAbTest( $this->getVal( 'ab', null ) );
```
Comment (slayful) [inline extensions/wikia/Search/WikiaSearchController.class.php:113]: The `namespaces` param was unused in the method. Removed to simplify.

File: extensions/wikia/Search/WikiaSearchController.class.php [hunk 2]
```diff
@@ -107,18 +110,148 @@ public function index() {
...
+
+		$this->setPageTitle( $searchConfig );
```
Comment (slayful) [inline extensions/wikia/Search/WikiaSearchController.class.php:115]: We can safely re-order this.

File: extensions/wikia/Search/classes/QueryService/Select/AbstractSelect.php [hunk 3]
```diff
@@ -327,6 +328,9 @@ protected function getRequestedFields() {
 			$fields[] = Utilities::field( $field, $language );
 		}
 
+		WikiaLogger::instance()->info("fields " .json_encode( $fields));
```
Comment (pwojcik86) [inline extensions/wikia/Search/classes/QueryService/Select/AbstractSelect.php]: Is this necessary?

File: extensions/wikia/Search/classes/QueryService/Select/Dismax/Video.php [hunk 4]
```diff
@@ -27,7 +27,7 @@ class Video extends OnWiki {
 		'canonical',
 		'host',
 		'ns',
-		'indexed',
+//		'indexed',
```
Comment (pwojcik86) [inline extensions/wikia/Search/classes/QueryService/Select/Dismax/Video.php]: Don't commit commented out code. Either remove it, or don't.

File: extensions/wikia/Search/classes/UnifiedSearch/UnifiedSearchService.php [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+				// normally, it would have been converted to ?namespace[]=1&namespace[]=0
+				'query' => build_query( $params, PHP_QUERY_RFC1738 ),
+			] );
+		}
+		catch ( ClientException $e ) {
+			WikiaLogger::instance()->error( "Failed to call unified-search", [
+				'error_message' => $e->getMessage(),
+				'status_code' => $e->getCode(),
+			] );
+			throw $e;
```
Comment (pwojcik86) [inline extensions/wikia/Search/classes/UnifiedSearch/UnifiedSearchService.php]: Is this exception handled in any way?

File: extensions/wikia/Search/classes/UnifiedSearch/SearchResultWrapper.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Wikia\Search\UnifiedSearch;
+
+use ArrayIterator;
+use Wikia\Search\Result;
+use Wikia\Search\SearchResultItems as SearchResultItemsAlias;
```
Comment (pwojcik86) [inline extensions/wikia/Search/classes/UnifiedSearch/SearchResultWrapper.php]: Why is this aliased?

**Review by pwojcik86 — CHANGES_REQUESTED** [review]

**Review by pwojcik86 — APPROVED** [review]

**Pull request merged**